### PR TITLE
First general clean-up

### PR DESCRIPTION
**Pull request by sjaeckel**
* Update doc a bit
* Fix build on Mac OS
* ~~Fix `dsa_make_params()`~~ #230 
* Improve math descriptors
* use `zeromem()` instead of `memset()` to clean prng buffers

### CODE REVIEW

File: src/headers/tomcrypt_math.h [hunk 1]
```diff
@@ -339,10 +355,15 @@ typedef struct {
        @param G   The point to multiply
        @param R   The destination for kG
        @param modulus  The modulus for the field
-       @param map Boolean indicated whether to map back to affine or not (can be ignored if you work in affine only)
+       @param map Boolean indicated whether to map back to affine or not
+                  (can be ignored if you work in affine only)
        @return CRYPT_OK on success
    */
-   int (*ecc_ptmul)(void *k, ecc_point *G, ecc_point *R, void *modulus, int map);
+   int (*ecc_ptmul)(     void *k,
```
Comment (karel-m) [inline src/headers/tomcrypt_math.h]: please avoid changing anything related to `ecc_*` especially when it is just a cosmetic change like this. It will cause a merge errors once I'll be trying to rebase `miko-ecc-enhancements` (I had to modify most of `ecc_*` functions in `tomcrypt_math.h`)

File: src/headers/tomcrypt_math.h [hunk 2]
```diff
@@ -352,7 +373,11 @@ typedef struct {
        @param mp   The "b" value from montgomery_setup()
        @return CRYPT_OK on success
    */
-   int (*ecc_ptadd)(ecc_point *P, ecc_point *Q, ecc_point *R, void *modulus, void *mp);
+   int (*ecc_ptadd)(ecc_point *P,
```
Comment (karel-m) [inline src/headers/tomcrypt_math.h]: please avoid changing anything related to `ecc_*` ... see above

File: src/headers/tomcrypt_math.h [hunk 3]
```diff
@@ -361,15 +386,20 @@ typedef struct {
        @param mp   The "b" value from montgomery_setup()
        @return CRYPT_OK on success
    */
-   int (*ecc_ptdbl)(ecc_point *P, ecc_point *R, void *modulus, void *mp);
+   int (*ecc_ptdbl)(ecc_point *P,
```
Comment (karel-m) [inline src/headers/tomcrypt_math.h]: please avoid changing anything related to `ecc_*` ... see above

File: src/headers/tomcrypt_math.h [hunk 4]
```diff
@@ -361,15 +386,20 @@ typedef struct {
...
+                    ecc_point *R,
+                         void *modulus,
+                         void *mp);
 
-   /** ECC mapping from projective to affine, currently uses (x,y,z) => (x/z^2, y/z^3, 1)
+   /** ECC mapping from projective to affine,
```
Comment (karel-m) [inline src/headers/tomcrypt_math.h]: please avoid changing anything related to `ecc_*` ... see above

File: src/headers/tomcrypt_math.h [hunk 5]
```diff
@@ -361,15 +386,20 @@ typedef struct {
...
+       currently uses (x,y,z) => (x/z^2, y/z^3, 1)
        @param P     The point to map
        @param modulus The modulus
        @param mp    The "b" value from montgomery_setup()
        @return CRYPT_OK on success
-       @remark  The mapping can be different but keep in mind a ecc_point only has three
-                integers (x,y,z) so if you use a different mapping you have to make it fit.
+       @remark The mapping can be different but keep in mind a
```
Comment (karel-m) [inline src/headers/tomcrypt_math.h]: please avoid changing anything related to `ecc_*` ... see above

File: src/headers/tomcrypt_math.h [hunk 6]
```diff
@@ -378,7 +408,7 @@ typedef struct {
        @param kA       What to multiple A by
        @param B        Second point to multiply
        @param kB       What to multiple B by
-       @param C        [out] Destination point (can overlap with A or B
+       @param C        [out] Destination point (can overlap with A or B)
```
Comment (karel-m) [inline src/headers/tomcrypt_math.h]: please avoid changing anything related to `ecc_*` ... see above

File: src/headers/tomcrypt_math.h [hunk 7]
```diff
@@ -15,6 +15,12 @@
    typedef void rsa_key;
 #endif
 
+#ifndef LTC_MILLER_RABIN_REPS
+   /* Number of repetitions of the Miller-Rabin test
```
Comment (karel-m) [inline src/headers/tomcrypt_math.h]: I would prefer to call it "rounds" instead of "repetitions".

**BUT** I think that all changes related to `is_prime`, M-R tests, missing Lucas etc. deserves a separate PR. I am not sure whether it is ready for 1.18

File: src/headers/tomcrypt_math.h (hunk 1 above)
Comment (sjaeckel) [inline src/headers/tomcrypt_math.h]: Simply rebase your stuff on develop (before merging this) and I'll take care of a rebase of `miko-ecc-enhancements` after merging this.

File: src/headers/tomcrypt_math.h (hunk 7 above)
Comment (sjaeckel) [inline src/headers/tomcrypt_math.h]: K, I'll move this to a separate PR

Regarding "repetitions" vs. "rounds", I don't really care, I took the wording from gmp and I somehow liked it.

File: src/headers/tomcrypt_math.h (hunk 1 above)
Comment (karel-m) [inline src/headers/tomcrypt_math.h]: I have rebased miko-ecc-enhancements on develop a minute ago.

**Comment by karel-m** [comment]
👍 please merge

File: src/headers/tomcrypt_math.h (hunk 1 above)
Comment (sjaeckel) [inline src/headers/tomcrypt_math.h]: Done
